src/battle_util.c

```diff
@@ -6098,12 +6098,18 @@ bool32 HasEnoughHpToEatBerry(u32 battlerId, u32 hpFraction, u32 itemId)
     return FALSE;
 }
 
+#if defined(ITEM_EXPANSION) && B_CONFUSE_BERRIES_HEAL >= GEN_7
```

Comment:
Hmm, I don't think the Item Expansion check is needed here.
Unlike the Sitrus Berry case where the HP value and method needed to match both in battle and in the Overworld, the HP threshold and HP healed here are 2 independent values.

If someone with only BE wants to use the Gen7+ values, they shouldn't need to come here and delete the IE check, as they should only need to tweak `battle_config.h` and the item data.

Reply:
Removed the check and expanded the config comment to explain.